tests/LanguageServerRename/OffsetExtractor.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Phpactor\Extension\LanguageServerRename\Tests;
+
+use Closure;
+
+class OffsetExtractor
+{
+    /** @var array */
+    private $points = [];
+    /** @var Closure[] */
+    private $pointCreateors = [];
+
+    /** @var array */
+    private $rangeOpenMarkers = [];
+    /** @var array */
+    private $rangeCloseMarkers = [];
+    /** @var Closure[] */
+    private $rangeCreateors = [];
+
+    public function registerPoint(string $name, string $marker, Closure $creator = null): void
```

Comment:
can we change this to not have a factory, but instead always return `ByteOffset` and `ByteOffsetRange` as relevant.